crates/ruma-common/src/events/poll/start.rs

```diff
@@ -0,0 +1,166 @@
+//! Types for the [`m.poll.start`] event.
+
+use std::convert::TryFrom;
+
+use js_int::{uint, UInt};
+use ruma_macros::EventContent;
+use serde::{Deserialize, Serialize};
+
+mod poll_answers_serde;
+
+use poll_answers_serde::PollAnswersDeHelper;
+
+use crate::{events::message::MessageContent, serde::StringEnum, PrivOwnedStr};
+
+/// The payload for a poll start event.
+#[derive(Clone, Debug, Serialize, Deserialize, EventContent)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+#[ruma_event(type = "org.matrix.msc3381.poll.start", alias = "m.poll.start", kind = MessageLike)]
+pub struct PollStartEventContent {
+    /// The poll start content of the message.
+    #[serde(rename = "org.matrix.msc3381.poll.start", alias = "m.poll.start")]
+    pub poll_start: PollStartContent,
+
+    /// Optional fallback text representation of the message, for clients that don't support polls.
+    #[serde(flatten, skip_serializing_if = "Option::is_none")]
+    pub message: Option<MessageContent>,
+}
+
+impl PollStartEventContent {
+    /// Creates a new `PollStartEventContent` with the given poll start content.
+    pub fn new(poll_start: PollStartContent) -> Self {
+        Self { poll_start, message: None }
+    }
+}
+
+/// Poll start content.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+pub struct PollStartContent {
+    /// The question of the poll.
+    pub question: MessageContent,
+
+    /// The kind of the poll.
+    #[serde(default)]
+    pub kind: PollKind,
+
+    /// The maximum number of responses a user is able to select.
+    ///
+    /// Defaults to `1`.
+    #[serde(default = "PollStartContent::default_max_selections")]
```

Comment:
Given it has a default, we should probably skip serializing the field if it has that value.